Makefile

```diff
@@ -202,19 +202,19 @@ test-docker-mount-code:
 	PACKAGES_DIR=$$(echo $$(pwd)/.venv/lib/python*/site-packages | awk '{print $$NF}'); \
 		DOCKER_FLAGS="$(DOCKER_FLAGS) --entrypoint= -v `pwd`/localstack/config.py:/opt/code/localstack/localstack/config.py -v `pwd`/localstack/constants.py:/opt/code/localstack/localstack/constants.py -v `pwd`/localstack/utils:/opt/code/localstack/localstack/utils -v `pwd`/localstack/services:/opt/code/localstack/localstack/services -v `pwd`/localstack/aws:/opt/code/localstack/localstack/aws -v `pwd`/Makefile:/opt/code/localstack/Makefile -v $$PACKAGES_DIR/moto:/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/ -e TEST_PATH=\\'$(TEST_PATH)\\' -e LAMBDA_JAVA_OPTS=$(LAMBDA_JAVA_OPTS) $(ENTRYPOINT)" CMD="make test" make docker-run
 
-lint:              		  ## Run code linter to check code style
-	($(VENV_RUN); python -m ruff check --show-source .)
+lint:              		  ## Run code linter to check code style and check if formatter would make changes
+	($(VENV_RUN); python -m ruff check --show-source . && python -m black --check .)
 
-lint-modified:     		  ## Run code linter on modified files
-	($(VENV_RUN); python -m ruff check --show-source `git diff --diff-filter=d --name-only HEAD | grep '\.py$$' | xargs` )
+lint-modified:     		  ## Run code linter to check code style and check if formatter would make changes on modified files
+	($(VENV_RUN); python -m ruff check --show-source `git diff --diff-filter=d --name-only HEAD | grep '\.py$$' | xargs` && python -m black --check `git diff --diff-filter=d --name-only HEAD | grep '\.py$$' | xargs`)
 
 check-aws-markers:     		  ## Lightweight check to ensure all AWS tests have proper compatibilty markers set
 	($(VENV_RUN); python -m pytest --co tests/aws/)
 
-format:            		  ## Run black and ruff code formatter
+format:            		  ## Run ruff and black to format the whole codebase
 	($(VENV_RUN); python -m ruff check --show-source --fix .; python -m black .)
 
-format-modified:   		  ## Run black and ruff code formatter on modified files
+format-modified:          ## Run ruff and black to format only modified code
 	($(VENV_RUN); python -m ruff check --show-source --fix `git diff --diff-filter=d --name-only HEAD | grep '\.py$$' | xargs`; python -m black `git diff --diff-filter=d --name-only HEAD | grep '\.py$$' | xargs` )
```

Comment:
Is the `;` between `ruff` and `lint` intentional compared to `&&` for `lint-modified`?
i.e., What's the reason for not failing the make target if `ruff` fails?

Reply:
That's a good point. It's basically just how it was previously (with isort).
When you execute `format`, you would like to run it on as much as possible (even if subcommands fail, f.e. because some code has syntax errors). The linting has to be strict, every single command has to trigger a non-zero exit code because it's the quality gate.